crates/hir_def/src/item_tree/lower.rs

```diff
@@ -42,6 +42,45 @@ pub(super) struct Ctx {
     forced_visibility: Option<RawVisibilityId>,
 }
 
+/// Returns `true` if the given intrinsic is unsafe to call or not.
+pub fn is_intrinsic_fn_unsafe(name: &Name) -> bool {
+    // Should be kept in sync with https://github.com/rust-lang/rust/blob/c6e4db620a7d2f569f11dcab627430921ea8aacf/compiler/rustc_typeck/src/check/intrinsic.rs#L68
+    *name != known::abort
```

Comment:
Let's write this as `[known::a, know::b, ...], contains(&name)`

Reply:
Other than that, lgmt!

I think it's fine to punt on attribute check

bors d+

Reply:
Oh, and the fn probably can be private? If it is, move it to the bottom of the file, as per https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#order-of-items